### ISSUE TITLE
Automatically grantRead for `apiKeySecretArn`

### ISSUE THREAD

**Issue by blimmer** [DataDog/datadog-cdk-constructs#174]
Currently, the documentation mentions that you have to remember to manually grant `GetSecretValue` when passing the API key as an ARN: https://github.com/DataDog/datadog-cdk-constructs/blob/8ffefd8dd5432805de312ade0f60d1015682b123/README.md#L175

It would be nice if this construct did it for you.

Instead of passing a secret arn string:

```ts
new Datadog(scope, 'Datadog', {
  apiKeySecretArn: 'arn:aws:secretsmanager:us-west-2:12345678910:secret:DATADOG_API_KEY'
});
```

If the user could pass an `ISecret`:

```ts
const secret = Secret.fromSecretPartialArn(this, 'DatadogApiKeySecret', 'arn:aws:secretsmanager:us-west-2:12345678910:secret:DATADOG_API_KEY');
new Datadog(scope, 'Datadog', {
  apiKeySecret: secret
});
```

Then, in the construct, you could `grantRead` for us:

```ts
apiKeySecret.grantRead(lambda)
```

**Comment by sfirrin** [comment]
Hi @blimmer thanks for this suggestion. We're going to work on implementing this and I will leave this ticket open for tracking

**Comment by blimmer** [comment]
Btw, this is made worse by the fact that the extension doesn't tell you clearly that you're missing this permission: https://github.com/DataDog/datadog-agent/issues/16927

**Comment by thedavl** [comment]
Hi @blimmer, we've implemented this feature and it has been included in the most recent `v2-1.7.0` release ([pypi](https://pypi.org/project/datadog-cdk-constructs-v2/), [npm](https://www.npmjs.com/package/datadog-cdk-constructs-v2)).

We're working on getting the associated docs changes merged, but in the meantime here's an example of how to use it (just like what you suggested):

```
const { Secret } = require('aws-cdk-lib/aws-secretsmanager');
const secret = Secret.fromSecretPartialArn(this, 'DatadogApiKeySecret' 'arn:aws:secretsmanager:us-west-1:123:secret:DATADOG_API_KEY');

const datadog = new Datadog(scope, 'Datadog', {
  ...
  apiKeySecret: secret
  ...
});
```

Thanks for your suggestion, please let me know if you've got any questions!
